addons/sale_timesheet/models/product.py

```diff
@@ -68,6 +68,17 @@ def _compute_product_tooltip(self):
                         "project for the order to track the time spent."
                     )
 
+    @api.onchange('type', 'service_type', 'service_policy')
+    def _update_uom_id(self):
+        for record in self:
+            if record.type == 'service' and record.service_type == 'timesheet' and \
+               (not record._origin or not record._origin.service_policy or record.service_policy != record._origin.service_policy):
```

Comment:
```suggestion
            if record.type == 'service' and record.service_type == 'timesheet' and \
               not (record._origin.service_policy and record.service_policy == record._origin.service_policy):
```
- Same comment as below
- "Mise en évidence" du not

Reply:
I'll update that ;)